### ISSUE TITLE
Control Plane does not scale beyond one node

### ISSUE THREAD

**Issue by marratj** [kubernetes-sigs/cluster-api-provider-vsphere#1096]
/kind bug

**What steps did you take and what happened:**
When setting the Control Plane replicas to a larger number than `1` (in `KubeadmControlPlane`) the control plane becomes ready but never scales beyond one node. The CAPV Manager seems to be not even trying to scale the number of control plane nodes up:

`"msg"="skipping reconcile when API server is online"  "reason"="controlPlaneInitialized"`

The same behavior exists no matter if a new cluster is created directly with 3 replicas set for `KubeadmControlPlane` or only scaled up to 3 after it has become Ready.

At the same time, the `cluster` resource is stuck at Scaling Up:

```
Status:
  Conditions:
    Last Transition Time:     2021-01-28T13:43:29Z
    Message:                  Scaling up control plane to 3 replicas (actual 1)
    Reason:                   ScalingUp
    Severity:                 Warning
    Status:                   False
    Type:                     Ready
```

**What did you expect to happen:**
When setting the number of replicas of `KubeadmControlPlane`, the number of control plane nodes/VMs should be scaled up properly by CAPV.

**Anything else you would like to add:**
The machines are stuck in `Provisioning` state, however, the Control Plane and cluster is up and running and perfectly reachable:

```
  Bootstrap Ready:  true
  Conditions:
    Last Transition Time:  2021-01-28T13:43:06Z
    Status:                True
    Type:                  Ready
    Last Transition Time:  2021-01-28T13:42:29Z
    Status:                True
    Type:                  BootstrapReady
    Last Transition Time:  2021-01-28T13:43:06Z
    Status:                True
    Type:                  InfrastructureReady
    Last Transition Time:  2021-01-28T13:56:31Z
    Reason:                NodeProvisioning
    Severity:              Warning
    Status:                False
    Type:                  NodeHealthy
  Infrastructure Ready:    true
  Last Updated:            2021-01-28T13:42:29Z
  Observed Generation:     3
  Phase:                   Provisioning
```

```
 kubectl get nodes 
NAME                                STATUS   ROLES    AGE   VERSION
vsphere-new-md-0-7966867cbd-52frp   Ready    <none>   20h   v1.18.2
vsphere-new-md-0-7966867cbd-r4vb5   Ready    <none>   20h   v1.18.2
vsphere-new-md-0-7966867cbd-tch8f   Ready    <none>   20h   v1.18.2
vsphere-new-xvtdj                   Ready    master   20h   v1.18.2
```

No error messages in the CAPV controller manager that would point to any issues with the vSphere environment.

**Environment:**

- Cluster-api-provider-vsphere version: `v0.7.4`
- Kubernetes version: (use `kubectl version`): `v1.18.2`

**Comment by marratj** [comment]
After some more digging, I found that the capi-manager is complaining about an empty providerID:

```
E0129 11:51:57.335710       1 machine_controller_noderef.go:152] controllers/Machine "msg"="Failed to parse ProviderID" "error"="providerID is empty" "providerID"={} "node"="vsphere-new-xvtdj"
```

**Comment by marratj** [comment]
Nevermind, I fixed the issue on our side. After reading some older, similar issues, I found that the CSI DaemonSet was not running as secrets were missing... but every time I created a new cluster I had the same issue, the ClusterResourceSet could not create the CSI config secret in the workload cluster.

Turns out that I shouldn't create the Cluster resources in the `cap-system` namespace... In a different namespace it worked just fine now.
